### ISSUE TITLE
libhtml-form-perl: t/form.t fails with HTTP::Message 7.00

### ISSUE THREAD

**Issue by gregoa** [libwww-perl/HTML-Form#50]
We have the following bug reported to the Debian package of
HTML-Form, c.f. https://bugs.debian.org/1084899

It doesn't seem to be a bug in the packaging, so you may want to take
a look. Thanks!

------8<-----------8<-----------8<-----------8<-----------8<-----

```
Source: libhtml-form-perl
Version: 6.11-1
Severity: serious
Tags: upstream ftbfs
Justification: fails to build from source

After upgrading HTTP::Message to 7.00, we see the following test
failure, probably caused by
https://github.com/libwww-perl/HTTP-Message/commit/968789db338f0c4f4756573614d26e4757d56deb


t/form.t ...............
ok 1
ok 2
ok 3
ok 4
ok 5
ok 6
ok 7
ok 8
ok 9
ok 10
ok 11
ok 12
ok 13
ok 14
not ok 15

ok 16
...


Cheers,
gregor

```
------8<-----------8<-----------8<-----------8<-----------8<-----

Thanks for considering,
  gregor herrmann,
  Debian Perl Group

**Comment by oalders** [comment]
Thanks, as always, @gregoa!

**Comment by oalders** [comment]
Culprit is https://github.com/libwww-perl/HTTP-Message/pull/196

**Comment by haarg** [comment]
Based on the HTML and URL specs, CRLF normalization is [part of the HTML spec](https://html.spec.whatwg.org/#converting-an-entry-list-to-a-list-of-name-value-pairs), not the [`application/x-www-form-urlencoded` format](https://url.spec.whatwg.org/#urlencoded-serializing). So I believe the change in HTTP-Message is correct, and HTML-Form's tests are correct, but the module needs to be updated to do the CRLF normalization that was previously happening in HTTP-Message.

**Comment by level420** [comment]
@oalders Just encountered this while building my perl docker image. Is/will this be solved somehow? Currently I'm installing HTML::Form forced to circumvent this problem.

**Comment by oalders** [comment]
@level420 we still need someone to fix this in `HTML::Form`. The current fix would be to pin `HTTP::Message` to a version < 7.00

**Comment by oalders** [comment]
Fixed via #53
